src/sources/logplex.rs

```diff
@@ -86,7 +56,56 @@ impl SourceConfig for LogplexConfig {
     }
 }
 
-fn body_to_events(body: impl Buf) -> Vec<Event> {
+fn decode_message(body: FullBody, header_map: HeaderMap) -> Result<Vec<Event>, ErrorMessage> {
+    //deal with headers
+    let msg_count = match usize::from_str(get_header(&header_map, "Logplex-Msg-Count")?) {
+        Ok(v) => v,
+        Err(e) => return Err(header_error_message("Logplex-Msg-Count", &e.to_string())),
+    };
+    let frame_id = get_header(&header_map, "Logplex-Frame-Id")?;
+    let drain_token = get_header(&header_map, "Logplex-Drain-Token")?;
+    info!(message = "Handling logplex request", %msg_count, %frame_id, %drain_token);
+
+    //deal with body
+    let events = body_to_events(body);
+
+    if events.len() != msg_count {
+        if cfg!(test) {
+            panic!("Parsed event count does not match message count header");
```

Comment:
Can we add the count information here too? I see it below!

We could probably just do the format once and let these share the message.

Reply:
Did this and the other suggestions in the new commit below
Thanks for the feedback 😄